syntaxes/src/build.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as fs from 'fs';
+
+import {GrammarDefinition, GrammarDefinitionValue, JsonObject, JsonObjectValue} from './types';
+
+import {template} from './template';
+
+const processValue = (value: GrammarDefinitionValue): JsonObjectValue => {
```

Comment:
Why not overload the function definition here rather than creating `GrammarDefinitionValue` in an external file?

Reply:
The main reason I created the type unions was to avoid having to break these onto multiple lines and to keep the call signatures a little tidier. It also, I think, makes the meaning a little easier to infer (we're transforming values that are valid in a grammar definition into values that are valid in a JSON object, which is maybe a little easier to grok than trying to discern the purpose of transforming `string|RegExp|GrammarDefinition|GrammarDefinition[]` into `string|JsonObject|JsonObject[]`). But abstracting it also kind of obfuscates it. I'm okay with changing it.

Reply:
Fixed!